py-polars/polars/series/series.py

```diff
@@ -6159,14 +6159,17 @@ def interpolate(self, method: InterpolationMethod = "linear") -> Series:
         ]
         """
 
-    def interpolate_by(self, by: IntoExpr) -> Series:
+    def interpolate_by(self, by: IntoExpr, *, extrapolate_flat: bool = False) -> Series:
         """
         Fill null values using interpolation based on another column.
 
         Parameters
         ----------
         by
             Column to interpolate values based on.
+        extrapolate_flat
+            If True, extrapolate the highest and lowest values of the expression in
+            the regions below and above the highest/lowest by values
```

Comment:
could we include an example which uses `extrapolate_flat`? users tend to learn better from examples than from descriptions